### ISSUE TITLE
Unlimited Rain drops are being created

### ISSUE THREAD

**Issue by kasvith** [sanjayaharshana/AnimTrap#13]
# Memory Management Bug

By looking at the code it can be guessed that unlimited amount of div elements will be created and after the usage they are not been destroyed. It can lead to memory errors of the user computer.

**It need to be fixed ASAP**

**Comment by kasvith** [comment]
# Workaround

One possible solution is to make rain drops disappear from the array itself after hitting the bottom of screen.

This can be done via an event, if this is possible i suggest you to stop creating new rain drops but use instead ones already in list to gain perfomance

**Comment by sanjayaharshana** [comment]
yeah.I'll admit that a lot of div tags is getting up here. I am going to be removed now.

**Comment by kasvith** [comment]
What about the workaround i mentioned?

**Comment by kasvith** [comment]
So what i mention is to render the div differently wrapped with an event

For example [look at this](https://stackoverflow.com/questions/18743144/jquery-event-listen-on-position-changed)

**Comment by sanjayaharshana** [comment]
We used to try to reduce javascript frequently some animation effect. so I will remove demo page rain effect and this code too.

**Comment by kasvith** [comment]
Well we need JS for create this, i suppose there is a CSS way for this. But it wont be supported by all browsers

[Look at this](https://codepen.io/Arhey/pen/NAQkwR)

**Comment by sanjayaharshana** [comment]
yeaah..that's a cool idea.

**Comment by kasvith** [comment]
Issue is still there
